pipelines/nextflow/modules/utils/utils.nf

```diff
@@ -50,3 +50,38 @@ def parse_list_param(String multi_value = '', List<String> allowed_values, Boole
     }
     return all_params
 }
+
+def generate_url(protocol, host, port, user, pass=null, database=null) {
+    // Generate a URL when all the credentials are provided
+    base_url = "${protocol}://${user}@${host}:${port}"
+
+    if (pass) {
+        base_url = "${protocol}://${user}:${pass}@${host}:${port}"
+    }
+    if (database) {
+        base_url += "/${database}"
+    }
+    return base_url
+}
+
+
+def extractMySQLArguments(urlString) {
+    //Extract the MySQL arguments from the url
+    def remove_protocol = urlString.split("//")[1]
```

Comment:
Can you add a check for the protocol (make sure it is indeed "mysql")? Also a check that the URL format is correct. If any of those are missing, it should throw an exception.